ui/apps/platform/src/Containers/Collections/CollectionsFormPage.tsx

```diff
@@ -22,27 +34,42 @@ function CollectionsFormPage({
     pageAction,
 }: CollectionsFormPageProps) {
     const isLargeScreen = useMediaQuery({ query: '(min-width: 992px)' }); // --pf-global--breakpoint--lg
-    const action = pageAction.type;
-    const collectionId = action !== 'create' ? pageAction.collectionId : undefined;
+    const collectionId = pageAction.type !== 'create' ? pageAction.collectionId : undefined;
     const collectionFetcher = useCallback(
         () => (collectionId ? getCollection(collectionId) : noopRequest),
         [collectionId]
     );
-    const { data } = useRestQuery(collectionFetcher);
+    const { data, loading, error } = useRestQuery<Collection, AggregateError>(collectionFetcher);
+
+    let content: ReactElement | undefined;
+
+    if (error) {
+        content = <>{/* TODO - Handle UI for parse errors */}</>;
+    }
+
+    if (loading) {
+        content = <>{/* TODO - Handle UI for loading state */}</>;
+    }
+
+    if (data) {
+        content = (
+            <CollectionForm
+                hasWriteAccessForCollections={hasWriteAccessForCollections}
+                action={pageAction}
+                initialData={data}
+                useInlineDrawer={isLargeScreen}
+                showBreadcrumbs
+                appendTableLinkAction={() => {
+                    /* TODO */
+                }}
+            />
+        );
+    }
```

Comment:
Although lint rules encourage independent `if` statements for `return` statement, what are pro and con of `if else if else if` structure for assignment, communicating to reader and TypeScript that cases are mutually exclusive? That is, if they are.

Reply:
Good call, they are not mutually exclusive, but this pattern would be a more accurate structure:
```typescript
if (error) {
} else if (data && !loading) {
} else if (loading) {
}
```

I'll update.